src/Stratis.Bitcoin.IntegrationTests/Mempool/MemoryPoolTests.cs

```diff
@@ -374,5 +376,62 @@ public void MempoolSyncTransactions()
                 TestHelper.WaitLoop(() => stratisNode2.CreateRPCClient().GetRawMempool().Length == 0);
             }
         }
+
+        [Fact]
+        public void MineBlocksBlockOrphanedAfterReorgTxsReturnedToMempool()
+        {
+            // Setup two synced nodes with some mined blocks.
+            string Password = "password";
+            string Name = "mywallet";
+            string AccountName = "account 0";
+            
+            NodeBuilder builder = NodeBuilder.Create(this);
+            CoreNode node1 = builder.CreateStratisPowNode(KnownNetworks.RegTest).NotInIBD().WithWallet();
+            CoreNode node2 = builder.CreateStratisPowNode(KnownNetworks.RegTest).NotInIBD().WithWallet();
+            MempoolValidationState mempoolValidationState = new MempoolValidationState(true);
+
+            builder.StartAll();
+
+            int maturity = (int)node1.FullNode.Network.Consensus.CoinbaseMaturity;
+            TestHelper.MineBlocks(node1, maturity + 20);
+            node1.CreateRPCClient().AddNode(node2.Endpoint, true);
+            TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(node1, node2));
+            
+            // Nodes disconnect.
+            node1.CreateRPCClient().RemoveNode(node2.Endpoint);
+
+            // Create tx and node 1 has this in mempool.
+            HdAddress receivingAddress = node2.FullNode.WalletManager().GetUnusedAddress(new WalletAccountReference(Name, AccountName));
+            Transaction transaction = node1.FullNode.WalletTransactionHandler().BuildTransaction(WalletTests.CreateContext(node1.FullNode.Network,
+                new WalletAccountReference(Name, AccountName), Password, receivingAddress.ScriptPubKey, Money.COIN * 100, FeeType.Medium, 101));
+
+            Assert.True(node1.FullNode.MempoolManager().Validator.AcceptToMemoryPool(mempoolValidationState, transaction).Result);
+            Assert.Contains(transaction.GetHash(), node1.FullNode.MempoolManager().GetMempoolAsync().Result);
+
+            // Node 2 has none in its mempool.
+            TestHelper.WaitLoop(() => node2.FullNode.MempoolManager().MempoolSize().Result == 0);
+
+            // Node 1 mines new tx into block - removed from mempool.
+            (HdAddress addressUsed, List<uint256> blockHashes) = TestHelper.MineBlocks(node1, 1);
+            uint256 minedBlockHash = blockHashes.Single();
+            TestHelper.WaitLoop(() => node1.FullNode.MempoolManager().MempoolSize().Result == 0);
+
+            // Node 2 mines two blocks to have greatest chainwork.
+            TestHelper.MineBlocks(node2, 2);
+
+            // Sync nodes and reorg occurs.
+            node1.CreateRPCClient().AddNode(node2.Endpoint, true);
```

Comment:
Use TestHelper.ConnectAndSync here remove line below.

Reply:
Needed to retain this as ConnectAndSync hangs as has no way of ignoring diffs across mempools as AddNode does.